runtime/vm/createramclass.cpp

```diff
@@ -2264,6 +2264,22 @@ internalCreateRAMClassDone(J9VMThread *vmThread, J9ClassLoader *classLoader, J9C
 			 */
 			const U_32 inheritedFlags = J9ClassHasWatchedFields | J9ClassIsExemptFromValidation | J9ClassHasIdentity | J9ClassEnsureHashed;
 			classFlags |= (superclass->classFlags & inheritedFlags);
+
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			if (J9_ARE_ALL_BITS_SET(classFlags, J9ClassHasIdentity)) {
```

Comment:
The spec has:
`if C is a value class and the class named as the direct superclass of C is neither the class Object nor a value class, loading throws an IncompatibleClassChangeError.`

A check on whether the super class is java.lang.Object should be added.

Reply:
java.lang.Object will not be marked with `J9ClassHasIdentity` since it has no superclass.

Reply:
> java.lang.Object will not be marked with J9ClassHasIdentity since it has no superclass.

See https://github.com/eclipse-openj9/openj9/pull/19716#discussion_r1643321687. In the old spec, java.lang.Object is neither identity nor value class. In the new spec, it is identity class.